pkg/table/tables/index.go

```diff
@@ -355,15 +354,6 @@ func (c *index) Create(sctx table.MutateContext, txn kv.Transaction, indexedValu
 			}
 			continue
 		}
-		if c.idxInfo.Global && len(value) != 0 && !bytes.Equal(value, idxVal) {
```

Comment:
Added by https://github.com/pingcap/tidb/pull/42311. After change the update operators, no need this code anymore.